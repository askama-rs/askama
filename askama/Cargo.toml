[package]
name = "askama"
version = "0.13.0"
description = "Type-safe, compiled Jinja-like templates for Rust"
keywords = ["markup", "template", "jinja2", "html"]
categories = ["template-engine"]
homepage = "https://askama.readthedocs.io/"
repository = "https://github.com/askama-rs/askama"
license = "MIT OR Apache-2.0"
edition = "2021"
rust-version = "1.81"

[package.metadata.docs.rs]
features = ["full"]
rustdoc-args = ["--generate-link-to-definition", "--cfg=docsrs"]

[[bench]]
name = "to-json"
harness = false
required-features = ["serde_json"]

[[bench]]
name = "escape"
harness = false

[dependencies]
itoa = "1.0.11"

# needed by feature "derive"
askama_derive = { version = "=0.13.0", path = "../askama_derive", default-features = false, optional = true }

# needed by feature "serde_json"
serde = { version = "1.0", optional = true, default-features = false }
serde_json = { version = "1.0", optional = true, default-features = false }

# needed by feature "urlencode"
percent-encoding = { version = "2.1.0", optional = true, default-features = false }

[dev-dependencies]
assert_matches = "1.5.0"
criterion = "0.5"

[badges]
maintenance = { status = "actively-developed" }

[features]
default = ["config", "derive", "std", "urlencode", "askama_derive?/default"]
full = ["default", "blocks", "code-in-doc", "serde_json", "askama_derive?/full"]

alloc = [
    "askama_derive?/alloc",
    "serde?/alloc",
    "serde_json?/alloc",
    "percent-encoding?/alloc",
]
blocks = ["askama_derive?/blocks"]
code-in-doc = ["askama_derive?/code-in-doc"]
config = ["askama_derive?/config"]
derive = ["askama_derive"]
serde_json = ["std", "askama_derive?/serde_json", "dep:serde", "dep:serde_json"]
std = [
    "alloc",
    "askama_derive?/std",
    "serde?/std",
    "serde_json?/std",
    "percent-encoding?/std",
]
urlencode = ["askama_derive?/urlencode", "dep:percent-encoding"]
