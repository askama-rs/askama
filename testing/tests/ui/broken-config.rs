use askama::Template;

#[derive(Template)]
#[template(source = "", ext = "txt", config = "no-such-config.toml")]
struct NoSuchConfig;

#[derive(Template)]
#[template(source = "", ext = "txt", config = "folder-config.toml")]
struct FolderConfig;

#[derive(Template)]
#[template(source = "", ext = "txt", config = "delim-clash.toml")]
struct DelimClash;

#[derive(Template)]
#[template(source = "", ext = "txt", config = "delim-too-short.toml")]
struct DelimTooShort;

#[derive(Template)]
#[template(source = "<+a+> and <+b+>", config = "operator-plus-config.toml", syntax = "plus", ext = "txt")]
struct PlusOperator;

#[derive(Template)]
#[template(source = "<)a(> and <)b(>", config = "operator-paren-config.toml", syntax = "paren", ext = "txt")]
struct ParenOperator;

fn main() {}
