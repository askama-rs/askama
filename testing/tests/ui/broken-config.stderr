error: `$WORKSPACE/target/tests/trybuild/askama_testing/no-such-config.toml` does not exist
 --> tests/ui/broken-config.rs:4:47
  |
4 | #[template(source = "", ext = "txt", config = "no-such-config.toml")]
  |                                               ^^^^^^^^^^^^^^^^^^^^^

error: unable to read $WORKSPACE/target/tests/trybuild/askama_testing/folder-config.toml: Is a directory (os error 21)
 --> tests/ui/broken-config.rs:8:47
  |
8 | #[template(source = "", ext = "txt", config = "folder-config.toml")]
  |                                               ^^^^^^^^^^^^^^^^^^^^

error: an opening delimiter may not be the prefix of another delimiter. The block delimiter ("<<<") clashes with the expression delimiter ("<<<<")
        --> testing/delim-clash.toml
  --> tests/ui/broken-config.rs:12:47
   |
12 | #[template(source = "", ext = "txt", config = "delim-clash.toml")]
   |                                               ^^^^^^^^^^^^^^^^^^

error: delimiters must be at least two characters long. The opening block delimiter ("<") is too short
        --> testing/delim-too-short.toml
  --> tests/ui/broken-config.rs:16:47
   |
16 | #[template(source = "", ext = "txt", config = "delim-too-short.toml")]
   |                                               ^^^^^^^^^^^^^^^^^^^^^^

error: closing delimiters may not start with operators. The closing expression delimiter ("+>") starts with operator `+`
        --> testing/operator-plus-config.toml
  --> tests/ui/broken-config.rs:20:49
   |
20 | #[template(source = "<+a+> and <+b+>", config = "operator-plus-config.toml", syntax = "plus", ext = "txt")]
   |                                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: closing delimiters may not start with operators. The closing expression delimiter ("(>") starts with operator `(`
        --> testing/operator-paren-config.toml
  --> tests/ui/broken-config.rs:24:49
   |
24 | #[template(source = "<)a(> and <)b(>", config = "operator-paren-config.toml", syntax = "paren", ext = "txt")]
   |                                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
